packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -221,10 +221,19 @@ export default util.createRule({
 
         if (expressionCount > 1) {
           if (previous.right.type === AST_NODE_TYPES.BinaryExpression) {
+            let operator = previous.right.operator;
+            if (
+              previous.right.operator === '!==' &&
+              previous.right.right.type === AST_NODE_TYPES.Literal &&
+              previous.right.right.raw === 'null'
```

Comment:
Just marking this todo for later...

```suggestion
              // TODO(#4820): Use the type checker to know whether this is `null`
              previous.right.right.type === AST_NODE_TYPES.Literal &&
              previous.right.right.raw === 'null'
```